.github/workflows/auto_publish_release.yml

```diff
@@ -0,0 +1,88 @@
+name: Release stage 2 - auto publish
+
+on:
+  pull_request:  # types AND paths
+    types: [closed]
+    paths: ['metomi/isodatetime/__init__.py']
+    # Don't use branches as we might create release on any branch
```

Comment:
Could a release happen accidentally if someone merges a PR that modifies the `__init__.py` file and the branch used was something like `prepare-to-update-python`?

Or maybe it could auto publish only from tag? Maybe the

```
on:
  release:
    created
```

I think `created` means that we can create branches with any names, changing any files, and it won't run. We can create a Draft Release too, and even upload to test-pypi if we want, and the action still won't run. And then if we mark the Release as final release (not a draft any longer), then the branch of that release would automatically be assigned for this action.

WDYT?

Reply:
The triggers GH provides are very course so Ronnie has put later checks in the workflow to do more advanced checking:

```
    if: github.event.pull_request.merged == true && startsWith(github.head_ref, 'prepare-')
```

Reply:
@kinow Oops, I might have overlooked that possibility... Having said that, if the version number isn't changed, the publish to PyPI.org step will fail, so everything should be alright.

Some possible solutions:
- Add `&& github.event.pull_request.user.id == 41898282` to the job's `if` condition, so it only triggers for the bot's PRs
- Create a new label - `release` - and have the 1st workflow add that to the PR. Then the 2nd workflow can use `&& contains(github.event.pull_request.labels.*.name, 'release')` in the `if` condition

Reply:
isodatetime doesn't have a lot of commits (it didn't at least, you've been doing a great work closing several issues from its backlog @MetRonnie), so that should be fine.

I asked because I think we should adopt it to Cylc Flow too; and also because I want to - shamelessly - copy it and use it to automate the release of the two modules I maintain in PYPI, and maybe port to another project that pushes to NPM :smile:

Reply:
Release label might be nice.

Reply:
And yes we definitely want to use this as a template for our other Python projects with as minimal modifications as possible.